include/zephyr/drivers/syscon.h

```diff
@@ -71,16 +71,16 @@ __subsystem struct syscon_driver_api {
  *
  * @param dev The device to get the register size for.
  * @param addr Where to write the base address.
- * @return 0 When addr was written to.
+ * @return 0 When @a addr was written to.
  */
 __syscall int syscon_get_base(const struct device *dev, uintptr_t *addr);
 
 static inline int z_impl_syscon_get_base(const struct device *dev, uintptr_t *addr)
 {
 	const struct syscon_driver_api *api = (const struct syscon_driver_api *)dev->api;
 
-	if (api == NULL) {
-		return -ENOTSUP;
+	if ((api == NULL) || (api->get_base == NULL)) {
```

Comment:
Do we actually expect anything to implement a `syscon` device driver but doesn't implement its APIs at all?